src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -359,6 +361,28 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  public SnapshotPreviewModel retrievePreview(UUID snapshotId, String tableName, Integer count) {
+    Snapshot snapshot = retrieve(snapshotId);
+
+    Optional<SnapshotTable> table =
+        snapshot.getTables().stream().filter(t -> t.getName().equals(tableName)).findFirst();
+
+    if (table.isEmpty()) {
+      throw new SnapshotPreviewException("No snapshot table exists with the name: " + tableName);
+    }
+
+    try {
+      List<Map<String, Object>> values = bigQueryPdao.getSnapshotTable(snapshot, tableName, count);
+
+      SnapshotPreviewModel previewModel = new SnapshotPreviewModel();
+      previewModel.setResult(Collections.singletonList(values));
+      return previewModel;
+    } catch (InterruptedException e) {
+      throw new SnapshotPreviewException(
+          "Error retrieving preview for snapshot " + snapshot.getName() + ": " + e);
```

Comment:
In general I prefer to include the caught exception when throwing, is there a reason to not include it here? It's possible it may include some helpful information.
```suggestion
          "Error retrieving preview for snapshot " + snapshot.getName(), e);
```